### PR TITLE
Refactoring Extended Time

### PR DESCRIPTION
**Pull request by Tang8330**
* Removing `NewKindDetailsFromTemplate`
* Renaming `TimestampTz`

### CODE REVIEW

File: lib/typing/ext/time.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ import (
 type ExtendedTimeKindType string
 
 const (
-	TimestampTzKindType  ExtendedTimeKindType = "timestamp_tz"
+	TimestampTZKindType  ExtendedTimeKindType = "timestamp_tz"
```
Comment (Tang8330) [inline lib/typing/ext/time.go:14]: Uppercase

File: lib/typing/ext/time.go [hunk 2]
```diff
@@ -28,8 +28,8 @@ var (
 		Format: RFC3339NanosecondNoTZ,
 	}
 
-	TimestampTz = NestedKind{
-		Type:   TimestampTzKindType,
+	TimestampTZ = NestedKind{
```
Comment (Tang8330) [inline lib/typing/ext/time.go:31]: Uppercase

File: lib/typing/typing.go [hunk 3]
```diff
@@ -92,15 +93,6 @@ func NewExtendedTimeDetails(details KindDetails, extendedType ext.ExtendedTimeKi
 	return details
 }
 
-func NewKindDetailsFromTemplate(details KindDetails, extendedType ext.ExtendedTimeKindType) KindDetails {
```
Comment (Tang8330) [inline lib/typing/typing.go:95]: Removed this function

File: lib/debezium/converters/time.go [hunk 4]
```diff
@@ -12,7 +12,7 @@ import (
 type Time struct{}
 
 func (Time) ToKindDetails() typing.KindDetails {
-	return typing.NewKindDetailsFromTemplate(typing.ETime, ext.TimeKindType)
+	return typing.NewExtendedTimeDetails(typing.ETime, ext.TimeKindType, "")
```
Comment (danafallon) [inline lib/debezium/converters/time.go:15]: Should we make this method take a receiver `t` and use `t.layout()` as the last arg here?

File: lib/optimization/table_data_merge_columns_test.go [hunk 5]
```diff
@@ -14,24 +14,24 @@ func TestTableData_UpdateInMemoryColumnsFromDestination_Tz(t *testing.T) {
...
 		assert.True(t, isOk)
-		assert.Equal(t, ext.TimestampTzKindType, updatedColumn.KindDetails.ExtendedTimeDetails.Type)
+		assert.Equal(t, ext.TimestampTZKindType, updatedColumn.KindDetails.ExtendedTimeDetails.Type)
 	}
 	{
 		// In memory is timestamp_ntz and destination is timestamp_tz
 		tableData := &TableData{inMemoryColumns: &columns.Columns{}}
-		kd := typing.NewKindDetailsFromTemplate(typing.ETime, ext.TimestampNTZKindType)
+		kd := typing.NewExtendedTimeDetails(typing.ETime, ext.TimestampNTZKindType, "")
 		kd.ExtendedTimeDetails.Format = ext.RFC3339MillisecondNoTZ
```
Comment (danafallon) [inline lib/optimization/table_data_merge_columns_test.go:28]: This can be passed into the constructor on the line above, right?

Comment (Tang8330) [reply lib/optimization/table_data_merge_columns_test.go:28]: Yes it can!

**Review by danafallon — APPROVED** [review]

**Pull request merged**